inference-engine/tests/functional/plugin/gna/preprocess_tests/precision_convert.cpp

```diff
@@ -0,0 +1,170 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "shared_test_classes/base/ov_subgraph.hpp"
+#include "ngraph_functions/utils/ngraph_helpers.hpp"
+#include "ngraph_functions/builders.hpp"
+#include "common_test_utils/common_utils.hpp"
+
+using namespace InferenceEngine;
+using namespace ov::test;
+
+namespace LayerTestsDefinitions {
+
+typedef std::tuple<
+        ov::element::Type,      // Net precision
+        ov::element::Type,      // Input precision
+        ov::element::Type,      // Output precision
+        std::string,            // Device name
+        std::map<std::string, std::string>  // Configuration
+> preprocessTestParamsSet;
+
+class PreprocessGNATest : public testing::WithParamInterface<preprocessTestParamsSet>,
+                                virtual public SubgraphBaseTest {
+public:
+    static std::string getTestCaseName(const testing::TestParamInfo<preprocessTestParamsSet>& obj) {
+        ov::element::Type net_type, in_type, out_type;
+        std::string target_device;
+        std::map<std::string, std::string> conf;
+
+        std::tie(net_type, in_type, out_type, target_device, conf) = obj.param;
+
+        std::ostringstream result;
+        result << "netPRC=" << net_type << "_";
+        result << "inPRC=" << in_type << "_";
+        result << "outPRC=" << out_type << "_";
+        result << "trgDev=" << target_device;
+        for (auto const &conf_i : conf) {
+            result << "_configItem=" << conf_i.first.c_str() << "_" << conf_i.second.c_str();
+        }
+        return result.str();
+    }
+protected:
+    void SetUp() override {
+        ov::element::Type net_type;
+        std::vector<InputShape> input_shapes = static_shapes_to_test_representation({{8, 2}, {8, 2}});
+        std::map<std::string, std::string> conf;
+        abs_threshold = std::numeric_limits<int32_t>::max();
+        rel_threshold = std::numeric_limits<int32_t>::max();
+
+        std::tie(net_type, inType, outType, targetDevice, conf) = this->GetParam();
```

Comment:
BTW, whether `inType`, `outType` are used?

Reply:
These are base class members and used to change the precision of inputs and outputs